app/pages/[teamSlug]/customers/[customerId].page.tsx

```diff
@@ -0,0 +1,91 @@
+import { useRouter } from 'next/router'
+import { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { Button } from '../../../frontend/components/button/button'
+import { InputField } from '../../../frontend/components/inputField/inputField'
+import { Modal } from '../../../frontend/components/modal'
+import { ProtectedPage } from '../../../frontend/components/protectedPage'
+import {
+  CustomerInput,
+  useCustomerDeleteMutation,
+  useCustomerQuery,
+  useCustomerUpdateMutation,
+} from '../../../frontend/generated/graphql'
+interface DeleteCustomerModalProps {
+  onClose: () => void
+}
+export const DeleteCustomerModal = ({ onClose }: DeleteCustomerModalProps) => {
+  const [, customerDelete] = useCustomerDeleteMutation()
+  const router = useRouter()
+  const { customerId, teamSlug } = router.query
+  const handleDeleteCustomer = async () => {
+    try {
+      await customerDelete({ customerId: customerId as string })
+
+      await router.push(`/${teamSlug}/customers`)
+    } catch {}
+  }
+  return (
+    <Modal
+      open={true}
+      onClose={onClose}
+      title={`Are you sure you want to delete project ${customerId}?`}
+      actions={
+        <>
+          <Button variant="secondary" onClick={onClose}>
+            {' '}
+            Cancel
+          </Button>
+          <Button variant="danger" onClick={handleDeleteCustomer}>
+            Delete
+          </Button>
+        </>
+      }
+    />
+  )
+}
+
+const ReportsPage = (): JSX.Element => {
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState<boolean>(false)
+  const router = useRouter()
+  const { customerId, teamSlug } = router.query
+  const [{ data, fetching }] = useCustomerQuery({ variables: { customerId: customerId?.toString() ?? '' } })
+  const [, updateCustomer] = useCustomerUpdateMutation()
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<CustomerInput>({ defaultValues: { title: data?.customer.title } })
+
+  if (fetching) {
+    return <div>Loading...</div>
+  }
```

Comment:
Similar to fetching, we should show an error message if the query could not load the customer.

Something like this:
```tsx
if (error) {
  return <div>Customer not found</div>
}
```